### ISSUE TITLE
[Feature Request] VDataTable: Ability to have grouped rows closed and/or open by default

### ISSUE THREAD

**Issue by webdevnerdstuff** [vuetifyjs/vuetify#17707]
### Problem to solve
Makes it more flexible for the users who want all rows open by default. This is the default for the v2 data table, but the v3 has them closed by default. The ability to choose which would be nice.

### Proposed solution
Perhaps a prop of `all-expanded` or `groups-expanded` (or whatever makes the most sense) with a `boolean` type of some kind?

**Comment by bcurless** [comment]
Yes this would be a nice feature. Adding an all open or all closed would be suitable for many people. but if someone was going to add this feature, they might as well make it so individual groups can be open or closed as well.

Personally, I want the ability to have them all expanded (open) by default. I would take a workaround to accomplish this at this point.

**Comment by dkmorgan** [comment]
> Personally, I want the ability to have them all expanded (open) by default. I would take a workaround to accomplish this at this point.

Is anyone aware of a suitable workaround to expand all groups by default?

**Comment by sw34** [comment]
I found this from [Stackoverflow](https://stackoverflow.com/questions/76111965/vuetify-3-v-data-table-open-groups-and-show-grouped-rows-by-default), it works but you cant collapse anymore, which for me is fine for now.


```
<template #group-header="{ item, columns, toggleGroup, isGroupOpen }">
      <tr>
        <td :colspan="columns.length">
          <VBtn
            :ref="
              (el) => {
                if (!isGroupOpen(item)) toggleGroup(item);
              }
            "
            size="small"
            variant="text"
            :icon="isGroupOpen(item) ? '$expand' : '$next'"
            @click="toggleGroup(item)"
          ></VBtn>
          {{ item.value }}
        </td>
      </tr>
    </template>
```

**Comment by dkmorgan** [comment]
@sw34 

You could try the updated solution which adds the use of the `loading` prop value to the condition to allow collapsing.
Unfortunately, it didn't work for me due to timing mismatches between the rendering of the groups and the `loading` prop value changes.

**Comment by sw34** [comment]
@dkmorgan 

I had the same timing issue. For my current use case, the current work around is good enough, but it would be nice to have either a better workaround or a permanent solution like the one proposed.

**Comment by crazyrabbit0** [comment]
I stumbled upon this problem today and came up with a solution that seems to work fine,
if anyone needs it, I posted it [here](https://github.com/vuetifyjs/vuetify/issues/17963#issuecomment-2078159451) and at [Stack Overflow](https://stackoverflow.com/a/78387199/24685914).

**Comment by philly-vanilly** [comment]
the proposed workarounds work only on visible rows. I have a table with multiple nested groups and the ones off-screen (nested or root) are ignored. I tried both v-data-table-virtual and v-data-table with :items-per-page="-1"

**Comment by laventnc** [comment]
incase anyone doesn't have a loading prop for data, you can also just do something like this:
```ts
const loaded = ref(false)
onMounted(() => {
  loaded.value = true
})
```
```vue
<template #group-header="{ item, columns, isGroupOpen, toggleGroup }">
  <template :ref="(el) => {
    if (!isGroupOpen(item) && !loaded) toggleGroup(item)
  }"></template>
</template>
```

**Comment by ar45** [comment]
Workaround https://stackoverflow.com/a/79331656/2059857

**Comment by mikadev** [comment]
Any update on this ? workaround is good but... why break the good thinks of v2 ?

**Comment by adamskeeled** [comment]
Yes, an optional prop to allow all groups to be expanded by default would be great! Currently the code that did this in v2 is currently commented out in v3: https://github.com/vuetifyjs/vuetify/blob/a3dc7bf42ded34dffbe24d19f020582377a7f5ff/packages/vuetify/src/components/VDataTable/composables/group.ts#L98L102

**Comment by mikadev** [comment]
as i explain i think we need to distinguish two different use case

1. collapse
2. group (like it was in v2 indeed) + adapt pagination accordingly + group must be on every page
